### PR TITLE
add a way to ask for project

### PR DESCRIPTION
**Pull request by FanchTheSystem**
- [x] Add a form in project page
- [x] Popup the form in dialog
- [x] Add Frontend management of project creation demand
- [x] Add Backend management of project creation demand
- [x] Squash Commit
- [x] Do some test

### CODE REVIEW

**Comment by osallou** [comment]
form should be shown only after user click on a "request project" kind button and be hidden after confirmation to avoid hiding projects info/details at the bottom of the page.
Or better could use a dialog.

**Comment by FanchTheSystem** [comment]
> form should be shown only after user click on a "request project" kind button and be hidden after confirmation to avoid hiding projects info/details at the bottom of the page.
> Or better could use a dialog.

Yes and it should be show opened when the user does have any project yet

**Comment by FanchTheSystem** [comment]
image

**Comment by FanchTheSystem** [comment]
I am testing it right now, but please @osallou start comment the code if you wish

After this pr, If you can add a tag, I will deliver this version on our production server.

**Comment by FanchTheSystem** [comment]
> After this pr, If you can add a tag, I will deliver this version on our production server.

Maybe you should wait a little before adding a tag, As I am fixing some minor bug in project workflow
https://github.com/genouest/genouestaccountmanager/pull/133
https://github.com/genouest/genouestaccountmanager/pull/132
https://github.com/genouest/genouestaccountmanager/pull/131
And maybe more are comming

**Comment by FanchTheSystem** [comment]
If it is ok for you, I am ready for a merge and a tag

File: app.js [hunk 1]
```diff
@@ -307,6 +307,7 @@ app.post('/project/:id', projects);
 app.post('/project/:id/request', projects);
 app.delete('/project/:id', projects);
 app.put('/project/:id/request', projects);
+app.post('/ask', projects);
```
Comment (osallou) [inline app.js]: Could you rename to more explicit route like
/ask/project or something like that

File: manager2/src/app/project/project.component.html [hunk 2]
```diff
@@ -1,160 +1,210 @@
...
+            <button type="button" class="btn btn-outline-primary" (click)="ask_for_project()" data-dismiss="modal">Ask Admin</button>
+            <button type="button" class="btn btn-outline-secondary" data-dismiss="modal">Cancel</button>
+          </div>
+        </div>
+      </form>
+    </div>
+
+  </div>
+</div>
+<button type="button" class="btn btn-primary" data-toggle="modal" data-target="#ask_project">Request A New Project</button>
```
Comment (osallou) [inline manager2/src/app/project/project.component.html]: Should show only if user is not admin

**Comment by osallou** [comment]
@FanchTheSystem  you  need to resync your code with latest updates.
Tests are failing on register. I tried to relaunch but fails on same test. Did you test with mocha locally?

**Comment by FanchTheSystem** [comment]
Yes I will merge master into my code to resolv conflict

**Comment by FanchTheSystem** [comment]
code synched and merged

File: manager2/src/app/project/project.component.html (hunk 2 above)
Comment (FanchTheSystem) [reply manager2/src/app/project/project.component.html]: done and tested

File: app.js (hunk 1 above)
Comment (FanchTheSystem) [reply app.js]: done and tested

**Pull request merged**